src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/ChannelsController.cs

```diff
@@ -294,5 +307,112 @@ public async Task<IActionResult> DeletePipelineFromChannel(int channelId, int pi
+            List<Microsoft.DotNet.Maestro.Client.Models.DefaultChannel> defaultChannels = (await barOnlyRemote.GetDefaultChannelsAsync()).ToList();
+
+            if (engLatestChannel != null)
+            {
+                defaultChannels.Add(
+                    new Microsoft.DotNet.Maestro.Client.Models.DefaultChannel(0, "https://github.com/dotnet/arcade", true)
+                    {
+                        Branch = "master",
+                        Channel = engLatestChannel
+                    }
+                );
+            }
+            if (eng3Channel != null)
+            {
+                defaultChannels.Add(
+                    new Microsoft.DotNet.Maestro.Client.Models.DefaultChannel(0, "https://github.com/dotnet/arcade", true)
+                    {
+                        Branch = "release/3.x",
+                        Channel = eng3Channel
+                    }
+                );
+            }
+
+            List<Microsoft.DotNet.Maestro.Client.Models.Subscription> subscriptions = (await barOnlyRemote.GetSubscriptionsAsync()).ToList();
+
+            // Build, then prune out what we don't want to see if the user specified
+            // channels.
+            DependencyFlowGraph flowGraph = await DependencyFlowGraph.BuildAsync(defaultChannels, subscriptions, barOnlyRemote, days);
+
+            string[] frequencies = includedFrequencies == default || includedFrequencies.Count() == 0 ? 
+                                   new string[] { "everyWeek", "twiceDaily", "everyDay", "everyBuild", "none", } : 
+                                   includedFrequencies.ToArray();
+
+            Microsoft.DotNet.Maestro.Client.Models.Channel targetChannel = null;
+
+            if (channelId != 0)
+            {
+                targetChannel = await barOnlyRemote.GetChannelAsync((int) channelId);
+            }
+
+            if (targetChannel != null)
+            {
+                flowGraph.PruneGraph(node => IsInterestingNode(targetChannel, node), edge => IsInterestingEdge(edge, includeDisabledSubscriptions, frequencies));
+            }
+
+            if (includeBuildTimes)
+            {
+                flowGraph.MarkBackEdges();
+                flowGraph.CalculateLongestBuildPaths();
+                flowGraph.MarkLongestBuildPath();
+            }
+
+            // Convert flow graph to correct return type
+            return Ok(FlowGraph.Create(flowGraph));
+        }
+
+        private bool IsInterestingNode(Microsoft.DotNet.Maestro.Client.Models.Channel targetChannel, DependencyFlowNode node)
+        {
+            return node.OutputChannels.Any(c => c == targetChannel.Name);
+        }
+
+        private bool IsInterestingEdge(DependencyFlowEdge edge, bool includeDisabledSubscriptions, string[] includedFrequencies)
+        {
+            if (!includeDisabledSubscriptions && !edge.Subscription.Enabled)
+            {
+                return false;
+            }
+            if (edge.Subscription.Policy == null)
```

Comment:
In what cases do we have a null policy?

Reply:
I was being paranoid. We probably don't. I think at one point, I wasn't including everything in the ToClientModel function, so added this in on the off chance that it didn't exist.